src/gm-apclientpp.cpp

```diff
@@ -684,3 +719,192 @@ double apclient_death_link(const char* cause)
         return GM_FALSE;
     return GM_BOOL(apclient && apclient->Bounce(data_j, {}, {}, { "DeathLink" }));
 }
+
+inline int key_to_int(const char* key) {
+    int i = 0;
+    while (*key) {
+        i *= 10;
+        i += *key++ - '0';
+    }
+    return i;
+}
+
+double apclient_json_proxy(const double proxy, const char* key)
+{
+    const std::lock_guard<std::mutex> lock(mut);
+    unsigned int int_proxy = (int)proxy;
+    try {
+        if (script_data[int_proxy].is_array() || script_data[int_proxy].is_binary()) {
+            int i = key_to_int(key);
+            if (i >= 0 && i < script_data[int_proxy].size())
```

Comment:
as above, this should be json::size_type. Note that the current code should give you a warning because you compare signed with unsigned.